File: example/src/test/java/ru/virgil/example/util/WithMockFirebaseUser.java
package ru.virgil.example.util;

import org.springframework.security.test.context.support.TestExecutionEvent;
import org.springframework.security.test.context.support.WithSecurityContext;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(
        factory = TestSecurityContextFactory.class,
        setupBefore = TestExecutionEvent.TEST_METHOD
)
public @interface WithMockFirebaseUser {

    String firebaseUserId() default "firebase-user-id";

    String firebaseAuthToken() default "firebase-auth-token";

}

File: examples/src/test/kotlin/ru/virgil/spring_tools/examples/integration/roles/user/WithMockFirebaseUser.kt
package ru.virgil.spring_tools.examples.integration.roles.user

import org.springframework.security.test.context.support.TestExecutionEvent
import org.springframework.security.test.context.support.WithSecurityContext
import ru.virgil.spring_tools.examples.security.SecurityUserAuthority

@Retention(AnnotationRetention.RUNTIME)
@WithSecurityContext(factory = MockUserSecurityContextFactory::class, setupBefore = TestExecutionEvent.TEST_METHOD)
annotation class WithMockFirebaseUser(
    val authorities: Array<SecurityUserAuthority> = [SecurityUserAuthority.ROLE_USER],
    val firebaseUserId: String = "user-id",
    val firebaseAuthToken: String = "user-auth-token",
)
